frontend/src/components/CytoscapeGraph/graphs/GraphStyles.ts

```diff
@@ -266,6 +266,9 @@ export class GraphStyles {
       if (node.hasWorkloadEntry) {
         badges = `<span class="${NodeIconWorkloadEntry} ${badgeMargin(badges)}"></span> ${badges}`;
       }
+      if (node.isGateway?.gatewayAPIInfo?.hostnames?.length !== undefined) {
+        badges = `<span class="${NodeIconGateway} ${badgeMargin(badges)}"></span> ${badges}`;
+      }
```

Comment:
This is probably OK but to totally avoid the chance of multiple gateway badges, maybe this should be incorporated into the `if else` directly below.  Something like:

```
     if (node.isRoot) {
        if (
          node.isGateway?.ingressInfo?.hostnames?.length !== undefined ||
          node.isGateway?.gatewayAPIInfo?.hostnames?.length !== undefined
        ) {
          badges = `<span class="${NodeIconGateway} ${badgeMargin(badges)}"></span> ${badges}`;
        }
        badges = `<span class="${NodeIconRoot} ${badgeMargin(badges)}"></span> ${badges}`;
      } else {
        if (
          node.isGateway?.egressInfo?.hostnames?.length !== undefined ||
          node.isGateway?.gatewayAPIInfo?.hostnames?.length !== undefined
        ) {
          badges = `<span class="${NodeIconGateway} ${badgeMargin(badges)}"></span> ${badges}`;
        }
      }
```

Actually, Are GatewayAPI gateways always ingress?  In that case eliminate the change in `else`.

Reply:
In my initial commit there was Gateway API under isRoot, then I changed to have it separate and independent of being isRoot edge, because Gateway API is always Ingress. Not sure if Gateway API should be always root edge or not.

Reply:
I think it's safe to assume that an ingress gateway will always also be a root (aka traffic source).  But if there is any doubt it can also be added to the `else` clause (now or later or never).